lib/model/avatar_url.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:path/path.dart';
+
+import '../log.dart';
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// DEFAULT_AVATAR_SIZE in zerver/lib/upload.py.
+const int defaultUploadSizePx = 100;
+
+/// MEDIUM_AVATAR_SIZE in zerver/lib/upload.py.
+const int mediumUploadSizePx = 500;
+
+abstract class AvatarURL {
+  static AvatarURL fromUserOrBotData({
+    String? rawAvatarUrl,
+    required Uri realm,
+    required int userId,
+    required String email,
+  }) {
+    if (rawAvatarUrl == null || rawAvatarUrl.isEmpty) {
```

Comment:
Why the special condition for an empty string?

In general the empty string is a string like other strings; it's not the same thing as `null` and generally shouldn't be treated the same.

Reply:
I've removed the check for null and empty string. As per the previous suggestion, if the resolvedUrl is null the SizedBox.shrink() widget is returned instead of recomputing the url using AvatarUrl.
[ea85867](https://github.com/zulip/zulip-flutter/pull/534/commits/ea858676f5d012214f4dc3113ba2df9540491d80)